inst/tests/tests.Rraw

```diff
@@ -7326,6 +7326,16 @@ test(1540.35, DT1[DT3, lapply(.SD, function(x) x * mul),
                  by=.EACHI, on=c(x="q", y="r"), roll=TRUE],
              DT1.copy[DT3, lapply(.SD, function(x) x * mul),
                  by=.EACHI, roll=TRUE])
+## more coverage tests for by = .EACHI, on = c(LHS = 'RHS') for numeric type
+set.seed(45L)
+DT1 = data.table(x=sample(letters[1:3], 15, TRUE), y=sample(6:10, 15, TRUE),
+                 a=sample(100, 15), b=runif(15))
+DT2 = CJ(x=letters[1:3], y=6:10)[, mul := sample(20, 15)][sample(15L, 5L)]
+DT3 = rbindlist(list(DT2, list(x="d", y=7L, mul=100L)))
+DT3 = DT3[sample(nrow(DT3))]
+DT1[ , x_num := match(x, letters) + .1]
+DT3[ , x_num := match(x, letters) + .1]
+test(1540.36, DT1[DT3, .(x_num), by=.EACHI, on=c(x_num="x_num")])
```

Comment:
strangely enough, I couldn't get this test to hit the `LGLSXP` branch:

```
DT1[ , x_lgl := x > = 'a']
DT3[ , x_lgl := x > 'a']
DT1[DT3, .(x_lgl), by=.EACHI, on=c(x_lgl="x_lgl")]
```

doesn't do it...